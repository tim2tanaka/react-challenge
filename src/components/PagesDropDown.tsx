import { ChangeEvent, useState, useEffect } from 'react';
import { actions } from '../app_state/actions';
import { useAppState, useAppDispatch } from '../app_state/context';
import { PaginationProps } from '../types';

export function DropdownList({loadData}: PaginationProps) {
  const state = useAppState();
  const dispatch = useAppDispatch();

  const [menuSelect, setMenuSelect] = useState(0);

  const { search, data, pagination } = state;

  useEffect(() => {
    if (search) loadData(search);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [menuSelect]);

  function handleChange(event: ChangeEvent<HTMLSelectElement>) {
    const pages = parseInt(event.target.value, 10);
    setMenuSelect(pages);
    dispatch({
      type: actions.setPagination.type,
      payload: {
        page: pagination.page,
        resultsPerPage: pages,
      },
    });
  }
  const dropdownListClass =
    search && data.length ? 'dropdownList' : 'dropdownList-hide';
  return (
    <div className={dropdownListClass}>
      {!!data.length && <select
        className="select-dropdown"
        value={pagination.resultsPerPage}
        onChange={(event) => handleChange(event)}
      >
        <option value={'5'}>5</option>
        <option value="10">10</option>
        <option value="15">15</option>
        <option value="20">20</option>
      </select>}
    </div>
  );
}
