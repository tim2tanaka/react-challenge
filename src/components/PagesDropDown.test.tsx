import { render } from '@testing-library/react';
import { DropdownList } from './PagesDropDown';

const loadData = jest.fn();
describe('PagesDropDown', () => {
  test('render PagesDropDown', () => {
    const { container } = render(<DropdownList loadData={loadData} />);
    expect(container).toMatchSnapshot();
  });
});
