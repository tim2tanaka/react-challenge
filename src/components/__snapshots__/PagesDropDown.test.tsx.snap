// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagesDropDown render PagesDropDown 1`] = `
<div>
  <div
    class="dropdownList-hide"
  />
</div>
`;
